dissect/formats/pcap.py

```diff
@@ -0,0 +1,475 @@
+                self.tsoffset = struct.unpack_from(fmt, opt.bytes)[0]
+        return ret
+
+class PCAPNG_ENHANCED_PACKET_BLOCK(PCAPNG_BLOCK_PARENT):
+    def __init__(self,  endian='little'):
+        PCAPNG_BLOCK_PARENT.__init__(self, endian)
+        self.blocktype   = uint32(endian=endian)
+        self.blocksize   = uint32(endian=endian)
+        self.interfaceid = uint32(endian=endian)
+        self.tstamphi    = int32(endian=endian)
+        self.tstamplow   = uint32(endian=endian)
+        self.caplen      = uint32(endian=endian).vsOnset(self.pcb_caplen)
+        self.packetlen   = uint32(endian=endian)
+        self.data        = vbytes(0)
+        self.options     = VArray([])
+
+    def pcb_caplen(self):
+        size = pad4bytes(self.caplen)
+        self['data'].vsResize(size)
+
+    def setPcapTimestamp(self, idb):
+        '''
+        Adds a libpcap compatible tvsec and tvusec fields, based on the pcapng timestamp
+        '''
+        #orange left off here
+        self.snaplen = idb.snaplen
+
+        tstamp = (self.tstamphi << 32) | self.tstamplow
+        scale = 1000000
+        if idb.tsresol is None:
+            #if not set, capture assumes 10e-6 resolution
+            pass
+        elif (0x80 & idb.tsresol) == 0:
+            # remaining bits are resolution, to a negative power of 10
+            scale = 10**(idb.tsresol & 0x7f)
+        else:
+            # remaining bits are resolution, to a negative power of 2
+            scale = 1 << (idb.tsresol & 0x7f)
+
+        self.tvsec = (tstamp / scale) + idb.tsoffset
+        self.tvusec = tstamp % scale
+
+class PCAPNG_SIMPLE_PACKET_BLOCK(VStruct):
+    '''
+    Note: no variable length options fields, so inheriting from vstruct directly
+    '''
+    def __init__(self, endian='little'):
+        VStruct.__init__(self)
+        self.blocktype  = uint32(endian=endian)
+        self.blocksize  = uint32(endian=endian).vsOnset(self.pcb_blocksize)
+        self.packetlen  = uint32(endian=endian)
+        self.data       = vbytes(0)
+        self.blocksize2 = uint32(endian=endian)
+
+    def pcb_blocksize(self):
+        self.caplen = pad4bytes(self.blocksize - 16)
+        self['data'].vsSetLength(self.caplen)
+
+    def setPcapTimestamp(self, idb):
+        #no timestamp in this type of block :(
+        self.tvsec = idb.tsoffset
+        self.tvusec = 0
+
+def iterPcapFileName(filename, reuse=False):
+    with open(filename, 'rb') as fd:
+        for x in iterPcapFile(fd, reuse=reuse):
+            yield x
+
+def iterPcapFile(fd, reuse=False):
+    '''
+    Figure out if it's a tcpdump format, or pcapng
+    '''
+    h = PCAP_FILE_HEADER()
+    b = fd.read(len(h))
+    h.vsParse(b, writeback=True)
+    fd.seek(0)
+    if h.magic == PCAPNG_BLOCKTYPE_SECTION_HEADER:
+        return _iterPcapNgFile(fd, reuse)
+    return _iterPcapFile(fd, reuse)
+
+
+def _iterPcapFile(fd, reuse=False):
+    h = PCAP_FILE_HEADER()
+    b = fd.read(len(h))
+    h.vsParse(b, writeback=True)
```

Comment:
why are we specifying writeback=True here ( and below ) when the object isn't backed by an FD or a memorymap?